selfdrive/ui/qt/onroad.cc

```diff
@@ -492,6 +492,12 @@ void AnnotatedCameraWidget::updateFrameMat() {
       .translate(-intrinsic_matrix.v[2], -intrinsic_matrix.v[5]);
 }
 
+float interp1d(float value, float start_min, float start_max, float end_min, float end_max) {
+  value = std::max(start_min, std::min(start_max, value));
+  float factor = (value - start_min) / (start_max - start_min);
+  return end_min + factor * (end_max - end_min);
+}
+
```

Comment:
todo: turn into equations and clamp until we have a common interp function. @adeebshihadeh do we have a need for this anywhere else?

Reply:
if it's generic, it should go in util.h, regardless if there's any current use cases

Reply:
🤦‍♂️ there's already a function in util that does exactly this!